app/src/main/java/be/huffle/allowanceplanner/activities/AddAllowanceActivity.java

```diff
@@ -39,42 +40,42 @@ protected void onCreate(Bundle savedInstanceState)
 		actionBar.setDisplayHomeAsUpEnabled(true);
 		actionBar.setTitle("Add Allowance");
 
-		addButton.setOnClickListener(this);
+		//addButton.setOnClickListener(this);
 	}
 
 	@Override
 	public boolean onOptionsItemSelected(@NonNull MenuItem item)
 	{
-		int id = item.getItemId();
-
-		if (id == android.R.id.home)
+		switch (item.getItemId())
 		{
-			finish();
-		}
-
-		return true;
-	}
-
-	@Override
-	public void onClick(View view)
-	{
-		switch (view.getId())
-		{
-			case R.id.add_button:
+			case android.R.id.home:
+				finish();
+				return true;
+			case R.id.add_menu_button:
 				String amountText = amountEditText.getText().toString();
 
 				if (amountText.isEmpty())
 				{
 					showDialog("Amount needs to be filled out");
-					break;
+					return true;
 				}
 
-				addAllowance(Float.valueOf(amountText),
+				addAllowance(Float.valueOf(amountEditText.getText().toString()),
 						descriptionEditText.getText().toString());
```

Comment:
This can possible be made slightly cleaner using an `if (...) {...} else {...}`.
Then you don't need the early return